### PR TITLE
OP-1150 OP-1151 OP-1153 OP-1154 OP-1155 Refactoring smells

### PR DESCRIPTION
**Pull request by Dharmil4602**
See OP-1150, OP-1151, OP-1153, OP-1154 and OP-1155.

This pull request addresses the "smell refactoring" by various refactoring techniques like Rename variable, move method. The refactoring effort aims to enhance code readability and maintainability by breaking down complex conditional statements into more modular and understandable components.

### Changes Made:

- Decomposed the conditional statements in AdmissionBrowserManager.java.
- Introduced new methods/functions to encapsulate logical conditions.
- Improved comments and documentation for better code understanding.
- Refactored SupplierBrowserManager.java, SupplierOperations.java, and related Tests.java file for better readability and debugging with Move Method.

Issue [Complex Conditional]: https://openhospital.atlassian.net/browse/OP-1150
Issue [Feature Envy]: https://openhospital.atlassian.net/browse/OP-1153

image

image

### CODE REVIEW

File: src/main/java/org/isf/admission/manager/AdmissionBrowserManager.java [hunk 1]
```diff
@@ -508,4 +507,8 @@ else if (dateOut.isBefore(ad.getAdmDate()) || dateOut.equals(ad.getAdmDate())) {
 			throw new OHDataValidationException(errors);
 		}
 	}
+	public boolean isAdmissionActiveOnDate(Admission admission, LocalDateTime checkDate) {
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/admission/manager/AdmissionBrowserManager.java:510]: I see little value in extracting the conditional into a method given the overhead that is required to invoke the method.  I don't believe this should be done.

File: src/test/java/org/isf/permissions/test/PermissionIoOperationsTest.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
+package org.isf.permissions.test;
```
Comment (dbmalkovsky) [inline src/test/java/org/isf/permissions/test/PermissionIoOperationsTest.java:1]: Missing license header in multiple files.

File: src/test/java/org/isf/permissions/test/PermissionIoOperationsTest.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+import org.isf.permissions.service.GroupPermissionIoOperationRepository;
+import org.isf.permissions.service.PermissionIoOperationRepository;
+import org.isf.permissions.service.PermissionIoOperations;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import java.util.List;
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (dbmalkovsky) [inline src/test/java/org/isf/permissions/test/PermissionIoOperationsTest.java:13]: The project standards do not support package level imports.   Please see the .ide-settings directory in this repo for import ordering and code formatting.

File: src/test/java/org/isf/permissions/test/PermissionIoOperationsTest.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+		permission.setGroupPermission(List.of(groupPermission1, groupPermission2));
+
+		when(permissionRepository.save(permission)).thenReturn(permission);
+		when(groupPermissionRepository.save(any())).thenReturn(groupPermission1);
+
+		// Act
+		Permission result = permissionIoOperations.insertPermission(permission);
+
+		// Assert
+		assertNotNull(result);
```
Comment (dbmalkovsky) [inline src/test/java/org/isf/permissions/test/PermissionIoOperationsTest.java:46]: The project standards are to use `aspectj`; please adopt the standard.

File: src/main/java/org/isf/dicom/manager/DicomLoader.java [hunk 5]
```diff
@@ -0,0 +1,434 @@
+package org.isf.dicom.manager;
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/dicom/manager/DicomLoader.java:1]: missing license header

File: src/main/java/org/isf/dicom/manager/SourceFiles.java [hunk 6]
```diff
@@ -56,12 +39,8 @@
...
 import org.springframework.util.StringUtils;
 
-import com.drew.imaging.ImageMetadataReader;
-import com.drew.imaging.ImageProcessingException;
-import com.drew.metadata.Metadata;
-import com.drew.metadata.MetadataException;
-import com.drew.metadata.exif.ExifDirectoryBase;
-import com.drew.metadata.exif.ExifIFD0Directory;
+import static org.isf.dicom.manager.DicomLoader.getSeriesDateTime;
+import static org.isf.dicom.manager.DicomLoader.getStudyDateTime;
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/dicom/manager/SourceFiles.java:43]: static imports are at the first ones in the list.

File: src/test/java/org/isf/utils/exception/OHExceptionTest.java [hunk 7]
```diff
@@ -0,0 +1,28 @@
+package org.isf.utils.exception;
```
Comment (dbmalkovsky) [inline src/test/java/org/isf/utils/exception/OHExceptionTest.java:1]: missing license header

**Comment by dbmalkovsky** [comment]
At a minimum please [Code Style](https://github.com/informatici/openhospital-core#code-style)  so there are fewer issues that we need to comment on.

**Comment by mwithi** [comment]
@Dharmil4602 please read my comment at https://openhospital.atlassian.net/browse/OP-1154?focusedCommentId=15187

**Comment by mwithi** [comment]
Closing for inactivity for too long time

**Pull request closed without merging**